internal/services/aws_reservation_service_test.go

```diff
@@ -0,0 +1,57 @@
+package services
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	ibClientStub "github.com/RHEnVision/provisioning-backend/internal/clients/image_builder/stubs"
+	sourcesClientStub "github.com/RHEnVision/provisioning-backend/internal/clients/sources/stubs"
+	"github.com/RHEnVision/provisioning-backend/internal/dao/stubs"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/RHEnVision/provisioning-backend/internal/testing/identity"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateAWSReservationHandler(t *testing.T) {
+	var err error
+	var json_data []byte
+	ctx := stubs.WithAccountDaoOne(context.Background())
+	ctx = identity.WithTenant(t, ctx)
+	ctx = sourcesClientStub.WithSourcesClient(ctx)
+	ctx = ibClientStub.WithImageBuilderClient(ctx)
+	ctx = stubs.WithReservationDao(ctx)
+	ctx = stubs.WithPubkeyDao(ctx)
+	pk := models.Pubkey{}
+	err = stubs.GeneratePubkey(ctx, &pk)
+	assert.Nil(t, err, fmt.Sprintf("Error GeneratePubkey: %v", err))
+
+	values := map[string]interface{}{
+		"source_id":     1,
+		"image_id":      "2bc640f6-927a-404a-9594-5b2da7e06608",
+		"amount":        1,
+		"instance_type": "t1.micro",
+		"pubkey_id":     pk.ID,
+	}
+	if json_data, err = json.Marshal(values); err != nil {
+		t.Fatal("unable to marshal values to json")
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", "/api/provisioning/reservations/aws", bytes.NewBuffer(json_data))
+	req.Header.Add("Content-Type", "application/json")
+	assert.Nil(t, err, fmt.Sprintf("Error creating a new request: %v", err))
+
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(CreateAWSReservation)
+	handler.ServeHTTP(rr, req)
```

Comment:
I seem to have a problem with an uninitialized job queue (https://github.com/RHEnVision/provisioning-backend/blob/42110d4edf1ac22a6fa4cfa110e21916b08ade2a/internal/services/aws_reservation_service.go#L133). 
 Do you any suggestion how should I stub it or initialize it?  @lzap , @ezr-ondrej

Reply:
I am not sure what level of integration do you expect in your test. I would say, that for unit tests, we should stub everything (database, REST clients, queue). So you want to only test the creation of job, the actual processing shall be tested separately in another test.

For this, you can utilize an implementation called "memory" which is asynchronous memory-only. You could initialize a single shared implementation in an `init()` function that would be common for all tests, with all handlers registered to "do nothing". It will silently swallow everything, maybe there could be some kind of counter that would increase or better (for concurrency reasons) providing a way to retrieve enqueued jobs.

You know what let me do a small prototype, I need to test my ideas.

Reply:
See in here: https://github.com/RHEnVision/provisioning-backend/pull/159